packages/core/src/browser/frontend-application.ts

```diff
@@ -224,6 +255,45 @@ export class FrontendApplication {
         }
     }
 
+    /**
+     * `beforeunload` listener implementation
+     */
+    protected getBeforeUnloadHandler() {
```

Comment:
i was thinking about something simple like:
```ts
    protected shouldStop(): boolean {
        const confirmExit = this.corePreferences['application.confirmExit'];
        if (confirmExit === 'never') {
            return false;
        }
        for (const contribution of this.contributions.getContributions()) {
            if (contribution.onWillStop) {
                if (contribution.onWillStop(this)) {
                    return true;
                }
            }
        }
        return confirmExit === 'always';
    }
```

I don't think a user wants to be overwhelmed by all reasons.

Reply:
I agree, but then it will be a bit misleading as not every `onWillStop` handlers will be called, wouldn't that be confusing?

Should we do two steps? `onWillStop(): void` and then `preventStop(): boolean` where we stop iterating on first `true` ?

I will go with what you propose until you share your thoughts